bee-ledger/src/types/balance_diff.rs

```diff
@@ -16,55 +12,26 @@ use std::collections::{
 
 /// Records a balance difference to apply to an address.
 #[derive(Clone, Debug, Default)]
-pub struct BalanceDiff {
-    amount: i64,
-    dust_allowance: i64,
-    dust_outputs: i64,
-}
+pub struct BalanceDiff(i64);
 
 impl BalanceDiff {
-    /// Creates a new `BalanceDiff`.
-    pub fn new(amount: i64, dust_allowance: i64, dust_outputs: i64) -> Result<Self, Error> {
+    /// Creates a new [`BalanceDiff`].
+    pub fn new(amount: i64) -> Result<Self, Error> {
         if amount.abs() as u64 > IOTA_SUPPLY {
             Err(Error::InvalidBalanceDiff(amount))
-        } else if dust_allowance.abs() as u64 > IOTA_SUPPLY {
-            Err(Error::InvalidBalanceDiff(dust_allowance))
-        } else if dust_outputs.abs() as u64 > IOTA_SUPPLY {
-            Err(Error::InvalidBalanceDiff(dust_outputs))
         } else {
-            Ok(Self {
-                amount,
-                dust_allowance,
-                dust_outputs,
-            })
+            Ok(Self(amount))
         }
     }
 
-    /// Returns the amount of a `BalanceDiff`.
+    /// Returns the amount of a [`BalanceDiff`].
     pub fn amount(&self) -> i64 {
-        self.amount
-    }
-
-    /// Returns the dust allowance of a `BalanceDiff`.
-    pub fn dust_allowance(&self) -> i64 {
-        self.dust_allowance
-    }
-
-    /// Returns the number of dust outputs of a `BalanceDiff`.
-    pub fn dust_outputs(&self) -> i64 {
-        self.dust_outputs
-    }
-
-    /// Returns whether dust allowance has been decreased or dust outputs has been increased.
-    pub fn is_dust_mutating(&self) -> bool {
-        self.dust_allowance < 0 || self.dust_outputs > 0
+        self.0
     }
 
-    /// Negates a `BalanceDiff`.
+    /// Negates a [`BalanceDiff`].
     pub fn negate(&mut self) {
```

Comment:
Yeah the crate will be refactored anyway and these need some due improvement. I will add a TODO for negate, good catch 👍

Reply:
Yeah it's technically bound to `IOTA_SUPPLY` but I don't know yet how this can change with shimmer and all, will leave a todo for now and convert to a simple comment later when I thought about it